plugins/woocommerce/src/Internal/FraudProtection/ApiClient.php

```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * ApiClient class file.
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\FraudProtection;
+
+use Automattic\Jetpack\Connection\Client as Jetpack_Connection_Client;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Handles communication with the WPCOM fraud protection endpoint.
+ *
+ * Uses Jetpack Connection for authenticated requests to the WPCOM endpoint
+ * to get fraud protection verdicts (allow, block, or challenge).
+ *
+ * This class implements a fail-open pattern: if the endpoint is unreachable,
+ * times out, or returns an error, it returns an "allow" decision to ensure
+ * legitimate transactions are never blocked due to service issues.
+ *
+ * @since 10.5.0
+ * @internal This class is part of the internal API and is subject to change without notice.
+ */
+class ApiClient {
+
+	/**
+	 * Default timeout for API requests in seconds.
+	 */
+	private const DEFAULT_TIMEOUT = 30;
```

Comment:
We'll need to update the timeout to a lower number if we don't want customers to be hanging for a long time in the checkout page.

To find a suitable time we'll need to benchmark the endpoint when it's ready.

Reply:
Issue created at WOOSUBS-1359